core/src/forest/Forest.h

```diff
@@ -40,6 +40,8 @@ class Forest {
   const std::vector<std::shared_ptr<Tree>>& get_trees() const;
   const size_t get_num_variables() const;
 
+  static Forest join(std::vector<std::shared_ptr<Forest>>& forests);
```

Comment:
It would be good to make this a `const` vector (I've been trying to encourage const correctness as much as possible).

Reply:
Great point!